controllers/windowsmachine_controller.go

```diff
@@ -110,13 +110,13 @@ func (r *WindowsMachineReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		// We need the create event to account for Machines that are in provisioned state but were created
 		// before WMCO started running
 		CreateFunc: func(e event.CreateEvent) bool {
-			return r.isValidMachine(e.Object) && isWindowsMachine(e.Object.GetLabels())
+			return isWindowsMachine(e.Object.GetLabels()) && r.isValidMachine(e.Object)
```

Comment:
Please add to your commit that the only visible advantage of this change is the debug messages not showing up.

Do we need to have a single helper for this?

Reply:
> Please add to your commit that the only visible advantage of this change is the debug messages not showing up.

.. and possible optimization of the busy time for the WindowsMachine controller since it spent cycles validating a non-windows machine, time which could be used to configure provisioned Windows Machine waiting in the queue.

Reply:
> Do we need to have a single helper for this?

Possibly, yes. @openshift/openshift-team-windows-containers  thoughts?

Reply:
>.. and possible optimization of the busy time for the WindowsMachine controller since it spent cycles validating a non-windows machine, time which could be used to configure provisioned Windows Machine waiting in the queue.

Have you measured the cycles? I doubt this has any actual effect given `isValidMachine()` just looks at memory and not making any API calls. While I am not against this change we should always keep the context in mind when optimizing. This is for future optimizations you are thinking about.

PS: I am not asking you to spend time profiling the method.

Reply:
I follow.